src/sentry/web/forms/accounts.py

```diff
@@ -361,6 +361,8 @@ def clean_username(self):
         value = re.sub(r"[ \n\t\r\0]*", "", value)
         if not value:
             return
+        if User.objects.filter(username__iexact=value).exclude(id=self.user.id).exists():
+            raise forms.ValidationError(_("An account is already registered with that username."))
```

Comment:
Hmm, this would actually introduce a user enumeration bug. Is there a more ambiguous way we could handle this or is the relocation feature sufficiently gated before we're at this step?

Reply:
The only way you'd get this link is if someone relocated an account with your email included as a user. I suppose a malicious actor could do this, but it would be very slow (~1 hour for feedback). I'd also note that today you can go into your sentry settings and change your username, with the exact same capabilities for user enumeration as presented here but without all of the ceremony of having to do a relocation first (https://sentry.sentry.io/settings/account/details/ if you want to try yourself).

Reply:
One thing we could do is rate limit this form, similar to what we do on L70 of this file. Wdyt?

Reply:
With the time effort and a rate-limit added to the form 👍 

I'll look into the "changing username" bug. :) Thank you!